src/python/pants/backend/scala/lint/scalafix/rules.py

```diff
@@ -261,15 +261,17 @@ def partition_info_for(
 @rule
 async def _scalafix_fix_partitions(
     request: ScalafixFixRequest.PartitionRequest[ScalafixFieldSet],
-) -> _ScalafixPartitionRequest:
-    return _ScalafixPartitionRequest(request.field_sets)
+) -> Partitions:
+    ret = await _partition_scalafix(_ScalafixPartitionRequest(request.field_sets), **implicitly())
+    return ret
 
 
 @rule
 async def _scalafix_lint_partitions(
     request: ScalafixLintRequest.PartitionRequest[ScalafixFieldSet],
-) -> _ScalafixPartitionRequest:
-    return _ScalafixPartitionRequest(request.field_sets)
+) -> Partitions:
```

Comment:
With polymorphic CBN we need a rule that has the polymorphic signature, with the same return value. So I had to slightly rejigger this in scalafix, which followed a slightly different idiom.